analoginput.ts

```diff
@@ -0,0 +1,48 @@
+//
+// *** adafruit_seesaw/adafruit_seesaw/analoginput.py ***
+//
+
+namespace analoginput {
+    // The MIT License (MIT)
+    // Copyright (c) 2017 Dean Miller for Adafruit Industries
+    // Permission is hereby granted, free of charge, to any person obtaining a copy
+    // of this software and associated documentation files (the "Software"), to deal
+    // in the Software without restriction, including without limitation the rights
+    // to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    // copies of the Software, and to permit persons to whom the Software is
+    // furnished to do so, subject to the following conditions:
+    // The above copyright notice and this permission notice shall be included in
+    // all copies or substantial portions of the Software.
+    // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    // THE SOFTWARE.
+    // pylint: disable=missing-docstring,invalid-name,too-many-public-methods
+    let __version__ = "0.0.0-auto.0"
+    let __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_seesaw.git"
+    export class AnalogInput {
+        _seesaw: Seesaw;
+        _pin: int8;
+        constructor(seesaw: Seesaw, pin: int8) {
+            this._seesaw = seesaw
+            this._pin = pin
+        }
+        
+        public deinit() {
+            ;
+        }
+        
+        get value(): uint16 {
+            return this._seesaw.analog_read(this._pin)
+        }
+        
+        get reference_voltage(): number {
```

Comment:
once the typescript build is clean, you can use VS Code "rename" refactoring to update the naming convention. It should be "referenceVoltage"